src/software/geom/polynomial.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <cmath>
+#include <vector>
+
+#include "software/geom/point.h"
+
+/**
+ * Polynomial is a representation of a polynomial
+ * of the form coeffs[0] + coeffs[1] * + coeff[2] * x^2...
```

Comment:
nit: IMO this is the "backwards, non-intuitive" way of storing and accepting coefficients as parameters. I would think to provide coefficients starting at the highest order, like I was writing it "left to right". @garethellis0 @EvanMorcom ?

Reply:
Yeah that would be nice, descending order is usually the convention

Reply:
I believe this is so we can easily prune the highest order 0 values (ie. if you give `x^2 + x^3 + 0*x^7` as `{0, 0, 1, 1, 0, 0, 0, 0}`, we should be able to shrink it to `{0, 0, 1, 1}`). Given that this is an internal implementation detail (see my comment above about why this should be with the member variable, not in the jdoc), I think it's fine.

Reply:
done

Reply:
no more pruning just give me a good vector